src/frontend/Typechecker.ml

```diff
@@ -1575,7 +1575,7 @@ and get_fn_decl_or_defn loc tenv id arg_tys rt body =
   match body with
   | {stmt= Skip; _} ->
       if exists_matching_fn_declared tenv id arg_tys rt then
-        Semantic_error.fn_decl_without_def loc |> error
+        Semantic_error.fn_decl_without_def loc id.name true |> error
```

Comment:
Is "re-declared without specifying a definition" really the best description of this error? Looks to me this happens when there are two identical declarations, regardless of whether there is also a definition later. I think "Function has already been declared. Expected a definition" would be more appropriate. (aka FunDeclExists, an error that does not seem to be in use currently?)

Reply:
Good catch. It seemed odd to me that these used the same error, but I didn't spot the one it should be using